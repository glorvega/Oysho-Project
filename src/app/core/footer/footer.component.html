<div class="footer-container">
    <div class="footer-container--row">
        <div class="footer-container--row__ayuda">
            <h5>Ayuda</h5>
            <ul *ngFor="let site of footerSites">
                <li *ngIf="site.group === 'Ayuda'">
                    <a href="{{site.url}}">{{site.title}}</a>
                </li>
            </ul>
        </div>

        <div class="footer-container--row__empresa">
            <h5>Empresa</h5>
            <ul class="footer-container--row__empresa--footer-links" *ngFor="let site of footerSites">
                <li *ngIf="site.group === 'Empresa'">
                    <a href="{{site.url}}">{{site.title}}</a>
                </li>
            </ul>
        </div>

        <div class="footer-container--row__links">
            <h5>Síguenos</h5>
            <ul class="footer-container--row__links--footer-links" *ngFor="let site of footerSites">
                <li *ngIf="site.group === 'Follow'">
                    <a href="{{site.url}}">{{site.title}}</a>
                </li>
            </ul>
        </div>

        <div class="footer-container--row__qr">
            <h5>Descarga nuestra app</h5>
            <img src="https://static.oysho.net/6/cms/assets/uploads/qrDownload_1.png?imwidth=200&ts=20221129021707"
                alt="código qr de Oysho" />
        </div>
    </div>
    <hr>
    <div class="footer-container__social">
        <div class="footer-container__social--copyright">
            <p>Copyright &copy; 2022 All Rights Reserved by
                <a href="https://github.com/glorvega">Gloria</a>.
            </p>
        </div>
    </div>
</div>