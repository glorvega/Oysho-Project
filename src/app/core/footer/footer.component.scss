.footer-container {
    display: flex;
    flex-direction: column;
    padding: 25px 15px 25px 15px;
    bottom: 0;
    background-color: #F9F9F9;
    font-size: 15px;
    line-height: 24px;
    color: #737373;
    position: fixed;
    width: 100%;

    &--row {
        display: flex;
        flex-direction: row;
        justify-content: space-around;

        &__ayuda {

            li {

                padding: 0;
            }
        }

        &__empresa {

            li {

                padding: 0;
            }
        }

        &__links {

            li {

                padding: 0;
            }
        }

        &__qr {

            display: flex;
            flex-direction: column;
            gap: 30px;
            align-items: center;

            img {
                min-height: auto;
                opacity: 1;
                width: 90px;
            }
        }

        h5 {
            color: #737373;
            font-size: 16px;
            text-transform: uppercase;
            letter-spacing: 2px
        }

        a {
            color: #737373;
            text-decoration: none;

            &:hover {
                color: #000;
                text-decoration: underline;
            }
        }

        .footer-links {

            li {
                display: block
            }

            a {
                color: #737373;

                &:hover {
                    color: #3366cc;
                    text-decoration: none;
                }
            }

            .inline {
                li {
                    display: inline-block
                }
            }
        }

    }

    hr {
        border-top-color: #bbb;
        opacity: 0.5;
        margin-top: 20px;
        margin-bottom: 20px;
        border: 0;
        border-top: 1px solid #eee;

        .small {
            margin: 20px 0
        }
    }
}