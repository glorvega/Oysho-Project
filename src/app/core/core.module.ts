import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NavbarComponent } from './navbar/navbar.component';
import { FooterComponent } from './footer/footer.component';
import { AppRoutingModule } from '../app-routing.module';
import { NavbarModule } from './navbar/navbar.module';
import { FooterModule } from './footer/footer.module';
import { SpinnerModule } from './spinner/spinner.module';
import { SpinnerComponent } from './spinner/spinner.component';

@NgModule({
  declarations: [],
  exports: [NavbarComponent, FooterComponent, SpinnerComponent],
  imports: [
    CommonModule,
    AppRoutingModule,
    NavbarModule,
    FooterModule,
    SpinnerModule,
  ],
})
export class CoreModule {}
