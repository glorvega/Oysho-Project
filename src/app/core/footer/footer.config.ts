import { FooterSites } from './interfaces/footer.interface';

export const FOOTER_SITES: FooterSites[] = [
  {
    url: 'https://www.oysho.com/es/contact.html',
    title: 'Contacto',
    group: 'Ayuda',
  },
  {
    url: 'https://www.oysho.com/es/att-client.html',
    title: 'Guía de compra',
    group: 'Ayuda',
  },
  {
    url: 'https://www.oysho.com/es/page/landing_guiatallas.html',
    title: 'Guía de tallas',
    group: 'Ayuda',
  },
  {
    url: 'https://www.oysho.com/es/page/landing_envios.html',
    title: 'Envíos y devoluciones',
    group: 'Ayuda',
  },
  {
    url: 'https://www.oysho.com/es/gift-card/home.html',
    title: 'Tarjeta regalo',
    group: 'Ayuda',
  },
  {
    url: 'https://www.oysho.com/integration/mectitofa/web/ticket?brand=OY&locale=es&market=ES',
    title: 'Solicitar factura',
    group: 'Ayuda',
  },
  {
    url: 'https://www.oysho.com/es/store-locator.html',
    title: 'Tiendas',
    group: 'Empresa',
  },
  {
    url: 'https://www.oysho.com/es/page/company.html',
    title: 'Sobre nosotros',
    group: 'Empresa',
  },
  {
    url: 'https://www.oysho.com/es/page/press.html',
    title: 'Prensa',
    group: 'Empresa',
  },
  {
    url: 'https://www.oysho.com/es/page/policies.html',
    title: 'Políticas',
    group: 'Empresa',
  },
  {
    url: 'https://static.oysho.net/6/static2/legal/cookiespolicy/CookiesPolicy_es.pdf?t=20221107015740',
    title: 'Configuración de cookies',
    group: 'Empresa',
  },
  {
    url: 'https://static.oysho.net/6/static2/legal/PrivacyPolicy/PrivPolicy_OY_ES_es.pdf?t=20221107015740',
    title: 'Política de privacidad',
    group: 'Empresa',
  },
  {
    url: 'https://www.facebook.com/oysho',
    title: 'Facebook',
    group: 'Follow',
  },
  {
    url: 'https://instagram.com/oysho',
    title: 'Instagram',
    group: 'Follow',
  },
  {
    url: 'https://www.youtube.com/c/oysho',
    title: 'YouTube',
    group: 'Follow',
  },
];
