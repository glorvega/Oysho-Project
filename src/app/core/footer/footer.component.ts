import { Component } from '@angular/core';
import { FOOTER_SITES } from './footer.config';
import { FooterSites } from './interfaces/footer.interface';

@Component({
  selector: 'app-footer',
  templateUrl: './footer.component.html',
  styleUrls: ['./footer.component.scss'],
})
export class FooterComponent {
  public footerSites: FooterSites[] = FOOTER_SITES;
}
