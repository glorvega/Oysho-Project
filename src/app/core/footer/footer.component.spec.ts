import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FooterComponent } from './footer.component';
import { FOOTER_SITES } from './footer.config';

describe('FooterComponent', () => {
  let component: FooterComponent;
  let fixture: ComponentFixture<FooterComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FooterComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(FooterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should render footer links', () => {
    const compiled = fixture.nativeElement;
    expect(compiled.querySelectorAll('a').length).toBe(FOOTER_SITES.length);
  });
  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
