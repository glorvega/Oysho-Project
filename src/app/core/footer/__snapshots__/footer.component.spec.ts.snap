// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should match snapshot 1`] = `
<app-footer
  __ngContext__={[Function Number]}
  footerSites={[Function Array]}
>
  <div
    class="footer-container"
  >
    <div
      class="footer-container__row"
    >
      <app-footer-links
        group="Ayuda"
      />
      <app-footer-links
        group="Empresa"
      />
      <app-footer-links
        group="Follow"
      />
      <div
        class="footer-container__row__qr"
      >
        <h5>
          Descarga nuestra app
        </h5>
        <img
          alt="código qr de Oysho"
          src="https://static.oysho.net/6/cms/assets/uploads/qrDownload_1.png?imwidth=200&ts=20221129021707"
        />
      </div>
    </div>
    <hr />
    <div
      class="footer-container__social"
    >
      <div
        class="footer-container__social--copyright"
      >
        <p>
          Copyright © 2022 All Rights Reserved by 
          <a
            href="https://github.com/glorvega"
          >
            Gloria
          </a>
          . 
        </p>
      </div>
    </div>
  </div>
</app-footer>
`;
